drivers/serial/uart_nrfx_uarte.c

```diff
@@ -1633,6 +1766,10 @@ static int uarte_nrfx_poll_in(const struct device *dev, unsigned char *c)
 		return -1;
 	}
 
+	if (IS_ENABLED(UARTE_ANY_CACHE) && (config->flags & UARTE_CFG_FLAG_CACHEABLE)) {
+		sys_cache_data_invd_range(config->poll_in_byte, 1);
```

Comment:
`poll_in_byte` could be managed by `dmm` to avoid manual cache management in this case. `dmm_buffer_in_prepare` would need to be called before starting the receiver and `dm_buffer_in_release` on ENDRX. But again, this looks like special case as this single byte is used internally by shim while `dmm` was rather designed for user-provided buffers